core/services/synchronization/web_socket_client.go

```diff
@@ -135,10 +156,12 @@ func (w *websocketClient) connectAndWritePump(parentCtx context.Context, wg *syn
 			defer cancel()
 
 			if err := w.connect(connectionCtx); err != nil {
+				w.connected = false
 				logger.Warn("Failed to connect to explorer (", w.url.String(), "): ", err)
 				break
 			}
 
+			w.connected = true
```

Comment:
We also need to reset `w.connected` to `false` if a disconnect happens after initial connection. This happens in the `writePump` after

```golang
wrapConnErrorIf(w.conn.Close()) // exclusive responsibility to close ws conn
```

It should also be at the bottom of `Close()`:

```
func (w *websocketClient) Close() error {
```

I think this file in particular could use another look for testability but that's outside the scope of your PR.

Reply:
Yeah, the tests to poke it in the right way aren't super clear.

Reply:
@dimroc I'm having trouble creating a test which executes the defer. I run into a timeout using this approach https://github.com/smartcontractkit/chainlink/pull/1328/files#diff-42e7b962a9af3a5edc897b07ec4e1cf0R161

Any tips?

I also converted the 2 bool fields to a single status